hazelcast/src/main/java/com/hazelcast/client/impl/proxy/ClientReliableTopicProxy.java

```diff
@@ -118,6 +138,21 @@ public void publish(@Nonnull E payload) {
         }
     }
 
+    @Override
+    public CompletionStage<E> publishAsync(@Nonnull E message) {
+        checkNotNull(message, NULL_MESSAGE_IS_NOT_ALLOWED);
+
+        Data element = toData(message);
+        ClientMessage request = TopicPublishCodec.encodeRequest(name, element);
+        try {
+            ClientInvocationFuture invocationFuture = new ClientInvocation(getClient(), request, getName()).invoke();
```

Comment:
I have got suspicious about this and deleted my comment. It seems, we are using partition id on ClientTopicProxy but not on ClientReliableTopicProxy. Order guarantees may not be given for Reliable one.  Can you still add tests for order guarantees for all of the new methods ?